src/main.ts

```diff
@@ -1,26 +1,43 @@
+import { getActionInputs, getOrganisationalUnits } from './helpers'
+import { ActionInput, WorkloadAccount } from './types'
+import { loadAccounts, writeAccounts } from './parser'
 import * as core from '@actions/core'
-import { wait } from './wait'
 
 /**
  * The main function for the action.
  * @returns {Promise<void>} Resolves when the action is complete.
  */
 export async function run(): Promise<void> {
   try {
-    const ms: string = core.getInput('milliseconds')
+    // Get the GH token and version file path
+    const inputs: ActionInput = getActionInputs([
+      { name: 'file_path', options: { required: true } },
+      { name: 'customer_id', options: { required: true } },
+      { name: 'spoc_email', options: { required: true } },
+      { name: 'organisational_units', options: { required: true } }
+    ])
 
-    // Debug logs are only output if the `ACTIONS_STEP_DEBUG` secret is true
-    core.debug(`Waiting ${ms} milliseconds ...`)
+    const accountDoc = loadAccounts(inputs.file_path)
 
-    // Log the current timestamp, wait, then log the new timestamp
-    core.debug(new Date().toTimeString())
-    await wait(parseInt(ms, 10))
-    core.debug(new Date().toTimeString())
+    for (const orgUnitName of getOrganisationalUnits(
+      inputs.organisational_units
+    )) {
+      accountDoc.workloadAccounts.push(
```

Comment:
I think we need to be a little defensive here and make sure that its unique, no customer already present with the same information?

Reply:
Agree, I think we want to ensure unique customer names and also unique email addresses as it can get the LZA config into a mess.

Reply:
Just making sure email address is unique is enough to protect against customer names too. Emails are also what LZA currently fails with.